core/shared/src/test/scala/fs2/ParEvalMapSuite.scala

```diff
@@ -268,4 +268,26 @@ class ParEvalMapSuite extends Fs2Suite {
       (1 to iterations).toList.as(action).sequence_
     }
   }
+
+  group("issue-2825, Stream shouldn't hang after exceptions thrown in") {
+    test("parEvalMapUnordered") {
+      Stream
+        .eval(IO.unit)
+        .parEvalMapUnordered(Int.MaxValue)(_ => throw new RuntimeException)
+        .compile
+        .drain
+        .attempt
+        .void
```

Comment:
maybe some `.timeout(2.seconds)`?

Reply:
Maybe. Munit already puts a 30 second timeout on each test, and I just followed the pattern of the test above which is checking a very similar issue 😆

Reply:
Oh, my bad, you are right it puts a timeout. 👍